src/modules/user/user.controller.ts

```diff
@@ -150,4 +156,18 @@ export class UserController {
   refreshToken(@UserDecorator() user: User) {
     return this.userUseCases.getAuthTokens(user);
   }
+
+  @Patch('password')
+  @ApiBearerAuth()
+  async updatePassword(
+    @RequestDecorator() req,
+    @Body() updatePasswordDto: UpdatePasswordDto,
+  ) {
+    try {
+      await this.userUseCases.updatePassword(req.user, updatePasswordDto);
+      return { status: 'success' };
+    } catch (err) {
+      throw new BadRequestException(err.message);
```

Comment:
I would check first which is the error rather than assuming that it is a bad exception always, internal server exceptions, as the more clear case, should not return the original message which may expose delicate details about something in terms of security.